pkg/systemd/logs.js

```diff
@@ -175,17 +175,18 @@ $(function() {
         function load_service_filters(match, options) {
             loading_services = true;
             current_services = new Set();
-            const service_options = Object.assign({ output: "verbose" }, options);
-            let cmd = journal.build_cmd(match, service_options)[0].join(" ");
-            cmd += " | grep SYSLOG_IDENTIFIER= | sort -u";
-            cockpit.spawn(["sh", "-ec", cmd], { host: options.host, superuser: "try" })
-                    .then(function(entries) {
-                        entries.split("\n").forEach(function(entry) {
+            const service_options = Object.assign({ output: "cat" }, options);
+            const cmd = journal.build_cmd(match, service_options)[0];
+            cmd.splice(1, 0, "--output-fields=SYSLOG_IDENTIFIER");
+            cockpit.spawn(cmd, { superuser: "try", err: "message" })
+                    .then(entries => {
+                        entries.split("\n").forEach(entry => {
                             if (entry)
-                                current_services.add(entry.substr(entry.indexOf('=') + 1));
+                                current_services.add(entry);
                         });
                     })
-                    .done(function () {
+                    .catch(e => console.log("Failed to load services:", e))
```

Comment:
Not that important, but should it be `console.warn()`? I'm not sure if our tests would then fail if the warning is hit, or if we only do that on console.error(). I suppose that depends on if we actually expect this to happen sometimes, e.g. on invalid user input?

Reply:
> I suppose that depends on if we actually expect this to happen sometimes, e.g. on invalid user input?

It can indeed happen on invalid input, like incorrect timestamp: `since:2021--23` will log `Failed to parse timestamp: 2021--23`

Reply:
I'll give it warning as users may actually notice they are doing something wrong